web/src/main/java/se/inera/intyg/webcert/web/csintegration/integration/CSIntegrationRequestFactory.java

```diff
@@ -538,11 +539,13 @@ public GetUnitQuestionsRequestDTO getUnitQuestionsRequestDTO(MessageQueryCriteri
             .build();
     }
 
-    public PatientCertificatesWithQARequestDTO getPatientCertificatesWithQARequestDTO(IntygWithNotificationsRequest request) {
-        return PatientCertificatesWithQARequestDTO.builder()
-            .personId(certificateServicePatientHelper.getPersonId(request.getPersonnummer()))
-            .careProviderId(request.getVardgivarId())
-            .unitIds(request.getEnhetId())
+    public CertificatesWithQARequestDTO getCertificatesWithQARequestDTO(List<Handelse> notifications) {
```

Comment:
Suggest that you change argument to list of certificateIds.

Also when getting the certificateIds from Handelse also add a distinct() so we don't send duplicate ids. Certificates will commonly have more than one Handelse.